### ISSUE TITLE
Invalid token `type` field for nullish coalescing operator

### ISSUE THREAD

**Issue by rubennorte** [babel/babel-eslint#631]
The `type` field in the token should be a string (as [per spec](https://eslint.org/docs/developer-guide/working-with-plugins#the-ast-specification)) but it's an object instead.

Example:

```javascript
parser.parse('foo ?? bar')
```

```
Node {
  type: 'Program',
  start: 0,
  end: 10,
  loc:
   SourceLocation {
     start: Position { line: 1, column: 0 },
     end: Position { line: 1, column: 10 } },
  range: [ 0, 10 ],
  comments: [],
  tokens:
   [ Token {
       type: 'Identifier',
       value: 'foo',
       start: 0,
       end: 3,
       loc: [SourceLocation],
       range: [Array] },
     Token {
       type: [BinopTokenType],
       value: '??',
       start: 4,
       end: 6,
       loc: [SourceLocation],
       range: [Array] },
     Token {
       type: 'Identifier',
       value: 'bar',
       start: 7,
       end: 10,
       loc: [SourceLocation],
       range: [Array] } ],
  sourceType: 'module',
  directives: undefined,
  body:
   [ Node {
       type: 'ExpressionStatement',
       start: 0,
       end: 10,
       loc: [SourceLocation],
       range: [Array],
       expression: [Node],
       _babelType: 'ExpressionStatement' } ] }
```

This was noticed by the ESLint folks in [this PR](https://github.com/eslint/eslint/pull/10476#issuecomment-397140501).

**Comment by rubennorte** [comment]
This is also an issue for the pipeline and optional chaining operators.
